subgraphs/stakedao/src/mappings/gaugeMappings.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  RewardPaid,
+  AddRewardCall,
+  Gauge as GaugeContract,
+  RewardAdded,
+} from "../../generated/templates/Gauge/Gauge";
+
+import * as utils from "../common/utils";
+import * as constants from "../common/constants";
+import { getUsdPriceOfToken } from "../modules/Price";
+import { RewardToken, Vault as VaultStore } from "../../generated/schema";
+import {
+  ethereum,
+  BigInt,
+  Address,
+  log,
+  BigDecimal,
+} from "@graphprotocol/graph-ts";
+
+export function handleRewardPaid(event: RewardPaid): void {
+  const gaugeAddress = event.address;
+  const gaugeContract = GaugeContract.bind(gaugeAddress);
+
+  let try_stakingToken = gaugeContract.try_stakingToken();
+
+  const vaultAddress = try_stakingToken.reverted
+    ? Address.fromString(constants.ZERO_ADDRESS)
+    : try_stakingToken.value;
+
+  const vault = VaultStore.load(vaultAddress.toHexString());
+  if (vault) {
+    let rewardTokenDecimals: BigInt, rewardTokenPrice: BigDecimal;
+    let rewardTokenEmissionsAmount: Array<BigInt> = [];
+    let rewardTokenEmissionsUSD: Array<BigDecimal> = [];
+
+    for (let i = 0; i < vault._rewardTokensIds.length; i++) {
+      let rewardToken = RewardToken.load(vault._rewardTokensIds[i]);
+      rewardTokenPrice = getUsdPriceOfToken(
+        Address.fromString(vault._rewardTokensIds[i])
+      );
+      rewardTokenDecimals = BigInt.fromI32(10).pow(rewardToken!.decimals as u8);
+
+      rewardTokenEmissionsAmount.push(event.params.reward);
+      rewardTokenEmissionsUSD.push(
+        rewardTokenPrice
+          .times(event.params.reward.toBigDecimal())
+          .div(rewardTokenDecimals.toBigDecimal())
+      );
+    }
+    vault.rewardTokenEmissionsAmount = rewardTokenEmissionsAmount;
+    vault.rewardTokenEmissionsUSD = rewardTokenEmissionsUSD;
+
+    vault.save();
+  }
+}
+
+export function handleAddReward(call: AddRewardCall): void {
+  const gaugeAddress = call.to;
+  const gaugeContract = GaugeContract.bind(gaugeAddress);
+
+  const vaultAddress = gaugeContract.stakingToken();
+  const vault = VaultStore.load(vaultAddress.toHexString());
+
+  if (vault) {
+    let rewardTokensIds: string[] = [];
+    let rewardTokenAddress: Address;
+    let try_rewardTokens: ethereum.CallResult<Address>;
+
+    for (let i = 0; i <= 10; i++) {

```

Comment:
looks like you are making an assumption that there will a max of 10 reward tokens? maybe a comment so other people know

Reply:
Okay I will do that